### ISSUE TITLE
Return number of spaxels in each bin

### ISSUE THREAD

**Issue by KateRowlands** [sdss/marvin#197]
Feature request:

I am trying to return the number of spaxels in each bin. At the moment I have to load each bin and count the number of spaxels which is very slow. It would be useful if there was a way to access the number of spaxels in each bin of a logcube.

**Comment by albireox** [comment]
Hi Kate,

Thanks for suggesting this. I agree that the current implementation of the bins is far from complete. We do want to implement better access to be bins, such as, from a cube, having a list of bins, each one of them containing the corresponding spaxels.

However, as you say, accessing so many spaxels makes loading slow so we still need to discuss and figure out the best way to implement this feature. If you have detailed ideas of how you'll like to see this implemented, we'd like to hear them.

In the meantime, if you are only interested in getting the number of spaxels for each bin, can I suggest you do something like this?

```
import numpy as np
from marvin.tools.maps import Maps
mm = Maps(plateifu='8485-1901', bintype='VOR10')
binid = mm['binid'].value
n_spaxels_bin = {ii: np.sum(binid == ii) for ii in sorted(np.unique(binid)) if ii > -1}
```

That will give you a dictionary in which each key is the binid and the associated value is the number of spaxels for that binid. The binids for a Maps are the same as for a logcube, so you can then apply them there. Does that help you?

**Comment by KateRowlands** [comment]
Hi Jose,

Thanks very much for the suggestion, this is very helpful. I didn't see your message for a while, so I managed to find another solution which produces the same result in a numpy array:

mm = Maps(plateifu=plateifu[i], bintype='VOR10')
binid = mm['binid'].value
bins_flat = binid.flatten()
spaxels_per_bin = np.bincount(bins_flat[np.where(bins_flat >-1)])

It turns out this is slightly faster than your method, but a dictionary might be a better format for the data.

I'm not exactly sure about how I'm going to work with my data and the bin IDs yet, I'll get back to you if I have any ideas.

Kate

**Comment by albireox** [comment]
Great! Yes, I can see how your method will also work. But this stresses that we need to rethink the implementation of Bin to make things more accessible.

**Comment by albireox** [comment]
MPL-6 may change the binning datamodel, which in turn may require a refactor of the `Bin` class. Let's wait until that decision is made before implementing one or other solution.

**Comment by albireox** [comment]
This is probably going to be fixed when #109 gets fixed, but making a note here just to be sure that's the case.
